File: webui/rockbox/src/Types/track.ts
export type Track = {
  id: string;
  trackNumber?: number | null;
  title: string;
  path?: string;
  artist: string;
  album?: string | null;
  time?: string | null;
  duration?: number | null;
  albumArt?: string | null;
  cover?: string | null;
  albumId?: string | null;
  artistId?: string | null;
  discnum?: number | null;
};

export type CurrentTrack = {
  id?: string;
  album?: string;
  artist?: string;
  title?: string;
  cover?: string;
  duration: number;
  progress: number;
  isPlaying?: boolean;
  albumId?: string | null;
  artistId?: string | null;
};

File: crates/traits/src/types/track.rs
#[derive(Debug, Clone, Default)]
pub struct Artist {
    pub id: String,
    pub name: String,
}

#[derive(Debug, Clone, Default)]
pub struct Album {
    pub id: String,
    pub title: String,
    pub cover: Option<String>,
}

#[derive(Debug, Clone, Default)]
pub struct Track {
    pub id: String,
    pub uri: String,
    pub path: String,
    pub title: String,
    pub artist: String,
    pub album: String,
    pub track_number: Option<u32>,
    pub disc_number: u32,
    pub duration: Option<f32>,
    pub album_artist: Option<String>,
    pub album_cover: Option<String>,
    pub album_id: Option<String>,
    pub artist_id: Option<String>,
}
